.github/workflows/build_directory_md.yml

```diff
@@ -0,0 +1,37 @@
+---
+name: build_directory_md
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    paths-ignore:
+      - '.github/**'
+      - '.scripts/**'
+      - 'DIRECTORY.md'
+
+jobs:
+  build_directory_md:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: jiro4989/setup-nim-action@v1
+
+      - name: Build file
+        run: |
+          git clean -f -x -d
+          nim r .scripts/directory.nim > DIRECTORY.md
+
+      - name: commit the file
+        run: |
+          date > generated.txt
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add DIRECTORY.md
+          git commit -m "Update DIRECTORY.md"
+          git push
```

Comment:
This is currently pushing directly to the `main` branch, which is good, however, if the branch is protected, there's no _safe_ way of allowing GitHub Actions to push directly. Thus, we need to create a pull request. We could use the GitHub CLI for that.

Let's discuss this on Discord so that you learn how to do the same. 🙂